ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -141,45 +85,29 @@ public void run() {
                     new CleanerCycleUpdater(MetricsConstants.COMPACTION_CLEANER_CYCLE_DURATION, startedAt));
           }
 
-          long minOpenTxnId = txnHandler.findMinOpenTxnIdForCleaner();
-
-          checkInterrupt();
-
-          List<CompactionInfo> readyToClean = txnHandler.findReadyToClean(minOpenTxnId, retentionTime);
+          for (Handler handler : handlers) {
+            List<CleaningRequest> readyToClean = handler.findReadyToClean();
```

Comment:
If the frst handler is broken for whatever reason, the remaining handlers won't get any chance to provide cleaning requests. This method call should be handled in a way to continue the handler iteration so subsquent ones can stil provide CleaningRequests

Reply:
Introduced try-catch block for every handler iteration. Done.